### ISSUE TITLE
Failing unit tests on Apple Silicon (ARM64)

### ISSUE THREAD

**Issue by mdklatt** [mdklatt/idea-netcdf-plugin#1]
Unit tests for UI components fail on Apple Silicon architecture. This is only a test failure. The plugin itself works as expected on Apple Silicon, and tests continue to pass on AMD64 platforms.

**Comment by mdklatt** [comment]
**SaveFileDialogTest.testConstructor**
```
[    924]   WARN - #c.i.j.JnaLoader - Unable to load JNA library (OS: Mac OS X 12.6.1)
java.lang.UnsatisfiedLinkError: Can't load library: /Users/mdklatt/Library/Caches/JNA/temp/jna14166039837381355669.tmp
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2633)
	at java.base/java.lang.Runtime.load0(Runtime.java:768)
	at java.base/java.lang.System.load(System.java:1837)
	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1018)
	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:988)
	at com.sun.jna.Native.<clinit>(Native.java:195)
	at com.intellij.jna.JnaLoader.load(JnaLoader.java:18)
	at com.intellij.jna.JnaLoader.isLoaded(JnaLoader.java:31)
	at com.intellij.openapi.util.io.FileSystemUtil.readCaseSensitivityByNativeAPI(FileSystemUtil.java:542)
	at com.intellij.openapi.util.io.FileSystemUtil.readParentCaseSensitivity(FileSystemUtil.java:460)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.generateCaseSensitivityChangedEventForUnknownCase(VfsImplUtil.java:316)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.updateCaseSensitivityIfUnknown(VirtualDirectoryImpl.java:314)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.findChild(VirtualDirectoryImpl.java:77)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.refreshAndFindChild(VirtualDirectoryImpl.java:346)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.refreshAndFindFileByPath(VfsImplUtil.java:122)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.refreshAndFindFileByPath(LocalFileSystemBase.java:69)
	at com.intellij.testFramework.HeavyPlatformTestCase.synchronizeTempDirVfs(HeavyPlatformTestCase.java:170)
	at com.intellij.testFramework.LightPlatformTestCase.initProject(LightPlatformTestCase.java:192)
	at com.intellij.testFramework.LightPlatformTestCase.lambda$doSetup$2(LightPlatformTestCase.java:277)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:490)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:523)
	at com.intellij.testFramework.LightPlatformTestCase.doSetup(LightPlatformTestCase.java:268)
	at com.intellij.testFramework.fixtures.impl.LightIdeaTestFixtureImpl.setUp(LightIdeaTestFixtureImpl.java:42)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$setUp$28(CodeInsightTestFixtureImpl.java:1242)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.java:33)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.setUp(CodeInsightTestFixtureImpl.java:1241)
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.setUp(BasePlatformTestCase.java:49)
	at dev.mdklatt.idea.netcdf.SaveFileDialogTest.setUp(FileDialogTest.kt:29)
	at com.intellij.testFramework.UsefulTestCase.invokeSetUp(UsefulTestCase.java:476)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:468)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:541)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:393)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
[   1127]   WARN - #c.i.i.s.e.StatisticsEventLogProviderUtil - Cannot find event log provider with recorder-id=FUS
[   1687]   WARN - #c.i.o.a.i.ActionManagerImpl - keymap "Visual Studio" not found PluginDescriptor(name=IDEA CORE, id=com.intellij, descriptorPath=plugin.xml, path=~/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2022.1/c91a19824ffcb28872ecd2761c83b250f703ff21/ideaIC-2022.1/lib, version=221.5080.210, package=null, isBundled=true)
[   1689]   WARN - #c.i.o.a.i.ActionManagerImpl - keymap "Eclipse" not found PluginDescriptor(name=IDEA CORE, id=com.intellij, descriptorPath=plugin.xml, path=~/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2022.1/c91a19824ffcb28872ecd2761c83b250f703ff21/ideaIC-2022.1/lib, version=221.5080.210, package=null, isBundled=true)
[   1689]   WARN - #c.i.o.a.i.ActionManagerImpl - keymap "NetBeans 6.5" not found PluginDescriptor(name=IDEA CORE, id=com.intellij, descriptorPath=plugin.xml, path=~/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2022.1/c91a19824ffcb28872ecd2761c83b250f703ff21/ideaIC-2022.1/lib, version=221.5080.210, package=null, isBundled=true)
[   1689]   WARN - #c.i.o.a.i.ActionManagerImpl - keymap "Visual Studio" not found PluginDescriptor(name=IDEA CORE, id=com.intellij, descriptorPath=plugin.xml, path=~/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2022.1/c91a19824ffcb28872ecd2761c83b250f703ff21/ideaIC-2022.1/lib, version=221.5080.210, package=null, isBundled=true)
[   1693]   WARN - #c.i.o.a.i.ActionManagerImpl - keymap "Eclipse (Mac OS X)" not found PluginDescriptor(name=IDEA CORE, id=com.intellij, descriptorPath=plugin.xml, path=~/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2022.1/c91a19824ffcb28872ecd2761c83b250f703ff21/ideaIC-2022.1/lib, version=221.5080.210, package=null, isBundled=true)

JNA library is not available
java.lang.AssertionError: JNA library is not available
	at com.intellij.ui.mac.foundation.Foundation.<clinit>(Foundation.java:31)
	at com.intellij.ui.mac.foundation.Foundation$NSAutoreleasePool.<init>(Foundation.java:481)
	at com.intellij.ui.components.MacScrollBarUI.callMac(MacScrollBarUI.java:200)
	at com.intellij.ui.components.MacScrollBarUI$Native.<init>(MacScrollBarUI.java:300)
	at com.intellij.ui.components.MacScrollBarUI$Style$1.<init>(MacScrollBarUI.java:251)
	at com.intellij.ui.components.MacScrollBarUI$Style.<clinit>(MacScrollBarUI.java:251)
	at com.intellij.ui.components.MacScrollBarUI.installUI(MacScrollBarUI.java:103)
	at java.desktop/javax.swing.JComponent.setUI(JComponent.java:685)
	at java.desktop/javax.swing.JScrollBar.setUI(JScrollBar.java:210)
	at com.intellij.ui.components.JBScrollBar.updateUI(JBScrollBar.java:87)
	at java.desktop/javax.swing.JScrollBar.<init>(JScrollBar.java:165)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:67)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:63)
	at com.intellij.ui.components.JBScrollPane.createVerticalScrollBar(JBScrollPane.java:279)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:300)
	at com.intellij.ui.components.JBScrollPane.<init>(JBScrollPane.java:95)
	at com.intellij.openapi.ui.DialogWrapper$ErrorText.<init>(DialogWrapper.java:2042)
	at com.intellij.openapi.ui.DialogWrapper.init(DialogWrapper.java:1259)
	at dev.mdklatt.idea.netcdf.SaveFileDialog.<init>(FileDialog.kt:19)
	at dev.mdklatt.idea.netcdf.SaveFileDialog.<init>(FileDialog.kt:13)
	at dev.mdklatt.idea.netcdf.SaveFileDialogTest.setUp(FileDialogTest.kt:30)
	at com.intellij.testFramework.UsefulTestCase.invokeSetUp(UsefulTestCase.java:476)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:468)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:541)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:393)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

**Comment by mdklatt** [comment]
**FileTabTest.testFileTab**
```
Could not initialize class com.intellij.ui.components.MacScrollBarUI$Style
java.lang.NoClassDefFoundError: Could not initialize class com.intellij.ui.components.MacScrollBarUI$Style
	at com.intellij.ui.components.MacScrollBarUI.installUI(MacScrollBarUI.java:103)
	at java.desktop/javax.swing.JComponent.setUI(JComponent.java:685)
	at java.desktop/javax.swing.JScrollBar.setUI(JScrollBar.java:210)
	at com.intellij.ui.components.JBScrollBar.updateUI(JBScrollBar.java:87)
	at java.desktop/javax.swing.JScrollBar.<init>(JScrollBar.java:165)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:67)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:63)
	at com.intellij.ui.components.JBScrollPane.createVerticalScrollBar(JBScrollPane.java:279)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:300)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:324)
	at com.intellij.ui.components.JBScrollPane.<init>(JBScrollPane.java:85)
	at dev.mdklatt.idea.netcdf.viewer.SchemaTab.<init>(Viewer.kt:164)
	at dev.mdklatt.idea.netcdf.viewer.FileTabTest.<init>(ViewerTest.kt:19)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:250)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:260)
	at org.junit.runners.BlockJUnit4ClassRunner$2.runReflectiveCall(BlockJUnit4ClassRunner.java:309)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

**Comment by mdklatt** [comment]
**FileTabTest.testDataTab**
``` 
Could not initialize class com.intellij.ui.components.MacScrollBarUI$Style
java.lang.NoClassDefFoundError: Could not initialize class com.intellij.ui.components.MacScrollBarUI$Style
	at com.intellij.ui.components.MacScrollBarUI.installUI(MacScrollBarUI.java:103)
	at java.desktop/javax.swing.JComponent.setUI(JComponent.java:685)
	at java.desktop/javax.swing.JScrollBar.setUI(JScrollBar.java:210)
	at com.intellij.ui.components.JBScrollBar.updateUI(JBScrollBar.java:87)
	at java.desktop/javax.swing.JScrollBar.<init>(JScrollBar.java:165)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:67)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:63)
	at com.intellij.ui.components.JBScrollPane.createVerticalScrollBar(JBScrollPane.java:279)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:300)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:324)
	at com.intellij.ui.components.JBScrollPane.<init>(JBScrollPane.java:85)
	at dev.mdklatt.idea.netcdf.viewer.SchemaTab.<init>(Viewer.kt:164)
	at dev.mdklatt.idea.netcdf.viewer.FileTabTest.<init>(ViewerTest.kt:19)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:250)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:260)
	at org.junit.runners.BlockJUnit4ClassRunner$2.runReflectiveCall(BlockJUnit4ClassRunner.java:309)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

**Comment by mdklatt** [comment]
It looks like all three tests are related to a problem loading native ARM64 UI components.  Not sure if *IJ* is fully native or using Rosetta on Silicon. The plugin itself works, so maybe the tests have to be configured to use the correct architecture. 

`SaveFileDialogTest` uses `com.intellij.testFramework.fixtures.BasePlatformTestCase`, but the `FileTab` tests do not. Do they need to? The tests work on AMD64 platforms as-is.

**Comment by mdklatt** [comment]
Per the [plugin development forum](https://intellij-support.jetbrains.com/hc/en-us/community/posts/8108214151186-Plugin-tests-are-failing-on-Apple-Silicon), the problem may be trying to test UI components outside of a [UI test](https://plugins.jetbrains.com/docs/intellij/testing-plugins.html#ui-tests). 

UI tests didn't seem necessary here because GUI interaction is not being tested, but maybe that's not the case. Even so, why do these tests work on AMD64 but not ARM64?

**Comment by mdklatt** [comment]
For now, disable these tests for `os.arch == aarch64`. This is easier said than done. Jupiter (JUnit5) includes annotations (*e.g.* `@DisableOnSystemProperty`) that are designed for this, but they do not work. The problem is that an exception is thrown when a UI component is initialized as as a text fixture property, but `@Disable*` only applies to test execution, not fixture instantiation.

For legacy JUnit tests with `BasePlatformTestCase`, the `@Disable*` or `@Ignore` (JUnit4) annotations are not supported at all.

**Comment by mdklatt** [comment]
The `UsefulTestCase` base class of `BasePlatformTestCase` provides a `shouldRunTest()` method. Commit b8d68ca56e4202ee73652a1a673329afc8c8f87a overrides this to skip tests on Apple Silicon. Notably, this also skips `setUp()`, which is actually where the error occurs. This may not be the proper way to do this, but it seems to work.

``` kotlin
internal class SaveFileDialogTest : BasePlatformTestCase() {
    /**
     * Determine if test should be run.
     *
     * @return: true if test should be run
     */
    override fun shouldRunTest(): Boolean {
        if (System.getProperty("os.arch") == "aarch64") {
            return false
        }
        return super.shouldRunTest()
    }

    // ...
}
```

**Comment by mdklatt** [comment]
All tests pass as expected when using a JUnit test runner (*i.e.* a JUnit Run Configuration in IntelliJ) instead of Gradle `test`.

**Comment by mdklatt** [comment]
The fact that all tests pass with JUnit but not Gradle seems like a possible issue with the IntelliJ SDK Gradle plugin.

[IDEA-304723](https://youtrack.jetbrains.com/issue/IDEA-304723/JUnit-tests-fail-with-Gradle-test-runner-on-Apple-Silicon)

**Comment by mdklatt** [comment]
This does not appear to be an IDE issue, so the community support forum is the appropriate places for this: [Unit tests fail on Apple Silicon with Gradle :test task but pass with JUnit test runner](https://intellij-support.jetbrains.com/hc/en-us/community/posts/8588167047954-Unit-tests-fail-on-Apple-Silicon-with-Gradle-test-task-but-pass-with-JUnit-test-runner).

**Comment by mdklatt** [comment]
Similar issue: [Minimal plugin test failing with "JNA library not available"](https://intellij-support.jetbrains.com/hc/en-us/community/posts/8438913861394-Minimal-plugin-test-failing-with-JNA-library-not-available-).

**Comment by mdklatt** [comment]
Commit 2701a9e63a7c055d6189413bd125930b70b84ee5 adds conditional test execution to *build.gradle.kts*. Broken tests are excluded by Gradle if the system architecture matches Apple Silicon (`aarch64`). This accomplishes three things:

1) Gradle builds use only the working subset of tests in an Apple Silicon dev environment.
2) An IntelliJ JUnit run configuration can be used to run all tests in any dev environment.
3) The GitHub Actions *Test* workflow runs all tests.

This workaround will allow other development to move forward until the broken tests can be fixed.

**Comment by mdklatt** [comment]
This is [fixed](eca267e037074b8cea439d693ea85b7b32f21e7b) with v1.13.3 of the SDK Gradle plugin.

**Comment by mdklatt** [comment]
Issue resolved by commit eca267e037074b8cea439d693ea85b7b32f21e7b.
